internal/controller/storageclient_controller.go

```diff
@@ -438,6 +439,9 @@ func (r *storageClientReconcile) deleteStorageClaims(claimName string) error {
 	storageClaim.Name = claimName
 
 	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); client.IgnoreNotFound(err) != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
```

Comment:
There is no reason to have another if here - as we already have an if filtering on errors.  
And there is no reason have a separate return here - as we already skipping storage claims that have no uid set

Just have the `IsNotMathError` check be part of the upper condition
(and also move to uses r.get instead r.Client.Get)
```
if err := r.Client.Get(storegeClaim); client.IgnoreNotFound(err) != nil  && !meta.IsNoMatchError(err) {
```